### PR TITLE
Handle soroban loadgen mode misconfig gracefully

### PR DESCRIPTION
**Pull request by ThomasBrady**
# Description

Resolves https://github.com/stellar/stellar-core/issues/4473

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/simulation/LoadGenerator.cpp [hunk 1]
```diff
@@ -348,6 +348,13 @@ LoadGenerator::start(GeneratedLoadConfig& cfg)
         if (cfg.modeInvokes())
         {
             auto const& sorobanLoadCfg = cfg.getSorobanConfig();
+            if (!mCodeKey)
```
Comment (SirTyson) [inline src/simulation/LoadGenerator.cpp:351]: Nit: Remove `mCodeKey` assert below.

**Review by SirTyson — DISMISSED** [review]
LGTM, super minor cleanup

File: src/simulation/test/LoadGeneratorTests.cpp [hunk 2]
```diff
@@ -125,6 +125,36 @@ TEST_CASE("generate load with unique accounts", "[loadgen]")
             10 * Herder::EXP_LEDGER_TIMESPAN_SECONDS, false);
     }
 }
+TEST_CASE("handle soroban loadgen mode misconfiguration gracefully",
```
Comment (SirTyson) [inline src/simulation/test/LoadGeneratorTests.cpp:128]: Nit: This doesn't have to be it's own test, but could just be a `SECTION` in "generate soroban load"

**Review by SirTyson — APPROVED** [review]
LGTM! Helpful change :)

**Review by SirTyson — APPROVED** [review]

**Comment by ThomasBrady** [comment]
> Unless I'm missing where we catch the exception, this change doesn't handle the failure gracefully, does it? The node still crashes. I think loadgen should gracefully fail and the node should continue operating normally (otherwise ssc will be stuck in pod restart loop without displaying any useful info)

Sorry you are right, this isn't "graceful", however I see there being two problems with this flow: the silent/misleading failure (`"Aborted (core dumped)"`) and gracefully propagating/handling the failure. This PR addressed the first part: if someone invokes without running setup they will now get an exception with the relevant error message. An exception is not exactly graceful however. I'm not sure how we can gracefully recover from this while also notifying the user.  `LoadGenerator::generateLoad` returns `void`; we could print an error and reset the load generator, but if the user is depending on this invocation to succeed, their test will fail regardless. Wouldn't it be better to raise an exception closer to the problematic action?

**Comment by marta-lokhova** [comment]
Load generator sets loadgen.run.failed on failure, which is used across our test suite and in supercluster (e.g. supercluster observed the metrics and fails the mission properly on failure)

**Pull request merged**